sdk/basyx/aasx.py

```diff
@@ -0,0 +1,1145 @@
+                raise KeyError(error_message)
+
+            object_store.add(identifiable)
+            ret.add(identifiable.id)
+
+    return object_store
+
+
+def _create_dict(data: ObjectStore) -> dict:
+    # separate different kind of objects
+    asset_administration_shells: List = []
+    submodels: List = []
+    concept_descriptions: List = []
+    for obj in data:
+        if isinstance(obj, model.AssetAdministrationShell):
+            asset_administration_shells.append(aas_jsonization.to_jsonable(obj))
+        elif isinstance(obj, model.Submodel):
+            submodels.append(aas_jsonization.to_jsonable(obj))
+        elif isinstance(obj, model.ConceptDescription):
+            concept_descriptions.append(aas_jsonization.to_jsonable(obj))
+    dict_: Dict[str, List] = {}
+    if asset_administration_shells:
+        dict_['assetAdministrationShells'] = asset_administration_shells
+    if submodels:
+        dict_['submodels'] = submodels
+    if concept_descriptions:
+        dict_['conceptDescriptions'] = concept_descriptions
+    return dict_
+
+
+class _DetachingTextIOWrapper(io.TextIOWrapper):
+    """
+    Like :class:`io.TextIOWrapper`, but detaches on context exit instead of closing the wrapped buffer.
+    """
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.detach()
+
+
+def write_aas_json_file(file: PathOrIO, data: ObjectStore, **kwargs) -> None:
+    """
+    Write a set of AAS objects to an Asset Administration Shell JSON file according to 'Details of the Asset
+    Administration Shell', chapter 5.5
+
+    :param file: A filename or file-like object to write the JSON-serialized data to
+    :param data: :class:`ObjectStore <basyx.aas.model.provider.AbstractObjectStore>` which contains different objects of
+                 the AAS meta model which should be serialized to a JSON file
+    :param kwargs: Additional keyword arguments to be passed to `json.dump()`
+    """
+
+    # json.dump() only accepts TextIO
+    cm: ContextManager[TextIO]
+    if isinstance(file, get_args(Path)):
+        # 'file' is a path, needs to be opened first
+        cm = open(file, "w", encoding="utf-8")
+    elif not hasattr(file, "encoding"):
+        # only TextIO has this attribute, so this must be BinaryIO, which needs to be wrapped
+        # mypy seems to have issues narrowing the type due to get_args()
+        cm = _DetachingTextIOWrapper(file, "utf-8", write_through=True)  # type: ignore[arg-type]
+    else:
+        # we already got TextIO, nothing needs to be done
+        # mypy seems to have issues narrowing the type due to get_args()
+        cm = contextlib.nullcontext(file)  # type: ignore[arg-type]
+    # serialize object to json#
+
+    with cm as fp:
+        json.dump(_create_dict(data), fp, **kwargs)
+
+
+def _write_element(file: PathOrBinaryIO, element: etree._Element, **kwargs) -> None:
+    etree.ElementTree(element).write(file, encoding="UTF-8", xml_declaration=True, method="xml", **kwargs)
```

Comment:
That method is just a wrapper and therefore provides little value to the readability of the code. Also it's only used once, so I'd incorporate this in the other method.